dashboard/app/controllers/report_abuse_controller.rb

```diff
@@ -66,7 +67,8 @@ def report_abuse
         return
       end
 
-      send_abuse_report(params[:name], params[:email], params[:age], params[:abuse_url])
+      username = current_user&.username
```

Comment:
Why do you check for `current_user` for `username` and not for the other properties like`name`, `email`, ...?

Reply:
Good question -- tl;dr I think user name isn't something we'd want to collect free form from a user, so grabbing it in Rails seemed like a reasonable approach. Some more detail on a little digging I did below!

This route is hit when submitting the form at studio.code.org/report_abuse. For some reason, we "round trip" name/email/age by including them when we render the form:

https://github.com/code-dot-org/code-dot-org/blob/8ce1272971018602185f0e6c289422a9d636bfd0/dashboard/app/controllers/report_abuse_controller.rb#L75-L83

Then return them when submitting the form (with the inputs for name/age/email hidden if we have them):

https://github.com/code-dot-org/code-dot-org/blob/8ce1272971018602185f0e6c289422a9d636bfd0/apps/src/code-studio/components/ReportAbuseForm.jsx#L102-L129

I think maybe we do this so that we can show fewer fields in the form if we already have the information, and collect it if we don't? A lot of this code is very old and originated here, which mentions hiding these fields:

https://github.com/code-dot-org/code-dot-org/pull/4023

Reply:
I could follow the same pattern as we use for `name` and pass the user name to front end, add a hidden input on the form, pass back to Rails. Feels kind of silly (I'd have to pass thru the user name as a prop) but maybe better to be consistent? Thoughts?

Reply:
TIL about 'round-tripping' in engineering. 😄

Reply:
lol googling suggests round tripping is a term I just made up, so don't quote me :) I like the name idea, let me take a quick look at that!

Reply:
Is this what you were thinking with 'round-tripping'? https://innoroo.com/blog/2019/05/17/round-trip-engineering-glossary/

Reply:
haha maybe? I think I just thought the phrase "round trip" described the situation well :)